src/snowflake/snowpark/dataframe.py

```diff
@@ -3285,6 +3300,22 @@ def _lateral(self, table_function: TableFunctionExpression) -> "DataFrame":
     def _show_string(self, n: int = 10, max_width: int = 50, **kwargs) -> str:
         query = self._plan.queries[-1].sql.strip().lower()
 
+        # TODO: A little hack to prevent infinite recursion.
+        if not "snowpark_coprocessor" in query:
+            # Add an Assign node that applies SpDataframeShow() to the input, followed by its Eval.
+            repr = self._session._ast_batch.assign()
+            repr.expr.sp_dataframe_show.id.CopyFrom(self._ast_stmt.var_id)
+            self._session._ast_batch.eval(repr)
+            
+            print(f'Original: {self._plan.queries}')
```

Comment:
nit: can we use logging? e.g., logging.debug or logging.trace.

Reply:
This is here temporarily. I wouldn't want to log these things for real.